### PR TITLE
Creating models 

### PR DESCRIPTION
**Pull request by mihrab34**
This branch is for building models with Active record

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  has_many :posts
+  validates :user_name, unique: true, presence: true,
```
Comment (mosaaleb) [inline app/models/user.rb]: - [ ] This line gives error, it's `uniqueness: true` not unique. Please check this out: [uniqueness_validation](https://guides.rubyonrails.org/active_record_validations.html#uniqueness)

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  has_many :posts
```
Comment (mosaaleb) [inline app/models/user.rb:2]: - [ ] To get a user comments, we need `has_many :comments` association.

Also, another advantage of adding comments association is that it enables you to build comments through association i.e.

```ruby
# building comment
Comment.create(user_id: 1, content: 'lorem ipsum')
# building comment through association (we don't need to assign user_id)
User.first.comments.create(content: 'lorem ipsum')
```

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+
+class Comment < ApplicationRecord
+  belongs_to :user, :post
```
Comment (mosaaleb) [inline app/models/comment.rb]: - [ ] We can't write multiple belongs to in the same line. This syntax raises an error in the console
      Please separate them: 
```ruby
belongs_to :user
belongs_to :post
```

**Review by mosaaleb — CHANGES_REQUESTED** [review]
#### Well done @mihrab34 :clap:

You have done most of the requirements for this project, still, there are some requirements needs to be handled before moving on:

- [ ] Linter is not set up for this repo, Let's add GitHub action and rubocop config file and make sure that we don't have any linting error. Check [this](https://github.com/microverseinc/linters-config/tree/master/ror) out.

- [ ] Take a look at my comments below

Please handle these requirements and submit a new code review.


##### Muhammad Ebeid

**Review by nikasulo — APPROVED** [review]
# Great work! @mihrab34 
Please merge and keep moving forward, I wish you the very best and see you at the top 🚀

**Pull request merged**